### PR TITLE
User management implemented : Fix #8

### LINKED ISSUE #8: Administration : UserListView/UserDetailView
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the UI-DSM
- [x] I have searched [open](https://github.com/RHEAGROUP/UI-DSM/issues) and [closed](https://github.com/RHEAGROUP/UI-DSM/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

The administrator should be able to manage users (Create/Delete/Update)
### System Configuration

- UI-DSM version:
  - [ ] UI-DSM:         
  - [ ] Other:              
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

### PR DESCRIPTION
**Pull request by antoineatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/UI-DSM/pulls) open
- [x] I have verified that I am following the UI-DSM [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/UI-DSM/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fix #8 

- [x] The administrator can add and delete user
- [x] The administrator can see details for specific user
- [x] Mvvm implementation refactor for previous components

### CODE REVIEW

**Comment by antoineatrhea** [comment]
image

File: UI_DSM/UI_DSM.Client/Program.cs [hunk 1]
```diff
@@ -40,15 +43,40 @@ public static async Task Main(string[] args)
...
+            await builder.Build().RunAsync();
+        }
+
+        /// <summary>
+        /// Register all ViewModels into the <see cref="WebAssemblyHostBuilder"/>
+        /// </summary>
+        /// <param name="builder">The <see cref="WebAssemblyHostBuilder"/></param>
+        private static void AddViewModels(WebAssemblyHostBuilder builder)
+        {
+            builder.Services.AddTransient<IUserManagementViewModel, UserManagementViewModel>();
```
Comment (lxatrhea) [inline UI_DSM/UI_DSM.Client/Program.cs:58]: Maybe do something generic/auto register? Find all i****ViewModel interfaces and their implementations and auto register those.

**Review by lxatrhea — APPROVED** [review]

File: UI_DSM/UI_DSM.Client/Program.cs (hunk 1 above)
Comment (antoineatrhea) [reply UI_DSM/UI_DSM.Client/Program.cs]: Implemented in the second commit, can you confirm that is what you though ?

**Review by lxatrhea — APPROVED** [review]

File: UI_DSM/UI_DSM.Client/Program.cs (hunk 1 above)
Comment (lxatrhea) [reply UI_DSM/UI_DSM.Client/Program.cs]: Yes, this is what I meant.

**Pull request merged**